### PR TITLE
feat: centralized log sanitization for sensitive data

### LINKED ISSUE #358: Remove private info from the debug logs
> Thanks @Fluimucil I edit your comment to remove the tokens and the installation number and the username, please take care when posting those messages. Thanks! 

 _Originally posted by @guerrerotook in [#355](https://github.com/guerrerotook/securitas-direct-new-api/issues/355#issuecomment-3982785487)_

### PR DESCRIPTION
**Pull request by clintongormley**
## Summary

- Adds a `SensitiveDataFilter` (Python `logging.Filter`) that automatically redacts sensitive data (auth tokens, refresh tokens, usernames, passwords, OTP data, installation numbers) from all log output under `custom_components.securitas`
- Filter is attached at `async_setup_entry` and removed at `async_unload_entry`; secrets are registered dynamically as they become available (login, token refresh, OTP validation, installation discovery)
- Installation numbers are partially masked (`***5678`), all other secrets are fully replaced with labels like `[AUTH_TOKEN]`, `[USERNAME]`, etc.
- Removes commented-out debug logging and raw tokens from exception messages

Closes #358

## Test plan

- [x] 12 unit tests covering redaction, dynamic updates, partial masking, edge cases
- [x] All 401 existing tests pass
- [x] ruff check/format clean

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: custom_components/securitas/__init__.py [hunk 1]
```diff
@@ -273,10 +284,15 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
...
+    if log_filter:
+        logging.getLogger("custom_components.securitas").removeFilter(log_filter)
+
     unload_ok = await hass.config_entries.async_unload_platforms(
         config_entry, PLATFORMS
     )
     hass.data[DOMAIN].pop(config_entry.entry_id, None)
+    hass.data[DOMAIN].pop("log_filter", None)
     if not hass.data[DOMAIN]:
         hass.data.pop(DOMAIN)
```
[the remark this answers is not in the document]
Comment (clintongormley) [reply custom_components/securitas/__init__.py]: Same as above — YAGNI. Single config entry integration. The unload path correctly removes the filter from all handlers and pops the key from hass.data[DOMAIN]. Multi-entry reference counting is not needed.

File: tests/test_log_filter.py [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    f = SensitiveDataFilter()
+    f.update_secret("password", "hunter2")
+
+    logger = logging.getLogger("test.securitas.filter_attach")
+    logger.addFilter(f)
+    logger.setLevel(logging.DEBUG)
+    handler = logging.StreamHandler(io.StringIO())
+    handler.setLevel(logging.DEBUG)
+    logger.addHandler(handler)
+
```
[the remark this answers is not in the document]
Comment (clintongormley) [reply tests/test_log_filter.py]: Added test_filter_on_handler_covers_child_loggers() — creates a parent logger with the filter on its handler, logs from a child logger, and verifies the output is redacted. This confirms handler-level filtering works for child loggers.
